### PR TITLE
File upload field type

### LINKED ISSUE #346: File upload field type
Supports use cases such as artist alley, cosplay competition etc.

Upload the file from the v2 Next.js backend to a configured location in Minio (S3) and store the URL (`s3://` or `https://`?) in the database. No need to mix Django FileField into this.

Sample field specification

```yaml
  - type: FileUpload
    required: true
    slug: avatar
    maxSizeKB: 1200
    maxFiles: 3
```

Eg.
```
BASE_URL/BUCKET_NAME/EVENT_SLUG/USE_CASE_ID/RESPONSE_ID/FIELD_SLUG-INDEX.EXT
https://minio.con2.fi/kompassi-uploads/form-responses/tracon2024/artist-alley/eaba89d6-bcea-4a02-9f4e-e59c80b4d6ea/header-1.png
```

Should the file be uploaded using a pre-signed URL directly from the browser to Minio? That would probably require a client component to perform the upload.

When the files are being consumed, need to provide pre-signed URLs for that. That the job of v1 or v2 backend?

File size limit? Does it need to be configurable across use cases?

Really big files (videos for AMV and cosplay backgrounds etc.) are a separate use case (integrate with xfer.velhosto.fi)

### PR DESCRIPTION
**Pull request by Hilzu**
Closes #346.

- [x] ~~The uploaded files are not public currently~~ The files shouldn't be public by default
- [x] Need to figure out how to configure MinIO URL so that backend and browser can communicate with it (proxy in backend?)
- [x] ~~Probably should deduce the mimetype from the filename in the mutation instead of trusting what the client passes in~~ This happens in Next.js backend so should be fine
- [x] Add prefix to files to be uploaded to make them unique (see pattern in the issue for inspiration)
- [x] Required attribute
- [x] ~~File size limit~~ Requires server-side validation. Implement later?
- [x] Upload multiple files
- [x] ~~GraphQL API access control?~~ https://github.com/con2/kompassi/issues/387

### CODE REVIEW

File: backend/forms/graphql/mutations/init_file_upload.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+s3_endpoint_url = settings.AWS_S3_ENDPOINT_URL
+s3_client = boto3.client(
+    "s3",
+    aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
+    aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY,
+    endpoint_url=s3_endpoint_url,
+)
+
+
+def generate_unique_id():
```
Comment (japsu) [inline backend/forms/graphql/mutations/init_file_upload.py:19]:
```
from uuid import uuid4
str(uuid4())
```
?

Comment (Hilzu) [reply backend/forms/graphql/mutations/init_file_upload.py:19]: UUIDv4 on hirveen pitkä ja ikävä

Comment (japsu) [reply backend/forms/graphql/mutations/init_file_upload.py:19]: 🤷‍♂️ ✅

**Comment by japsu** [comment]
Suggestion:

1. Split `FileUpload` into `SingleFileUpload` and `MultiFileUpload`. 
2. In `formData`, transmit both always as `string[]` of S3 URLs.
3. In `process_form_data`, turn `SingleFileUpload` value into `string` (warn if more than 1) and leave `MultiFileUpload` value as `string[]`.

This way, `SingleFileUpload` and `MultiFileUpload` would have consistent types and be analoguous to `SingleSelect` and `MultiSelect`.

Alternative 2: As-is. Consumer needs to always check if it's `string` (one file) or `string[]` (multiple files).

Alternative 3: Always represented in `form_data` and `values` as `string[]`. Consumer is responsible for checking if there are more than 1 if it makes a difference to the consumer. If represented visually, probably doesn't make a difference; if processed automatically, sometimes it may be that There Can Only Be One.

No strong opinion except for minor dislike for 2 on the grounds that consistent types are nice. 3 likely to be less work than 1.

For further work (TODO @japsu explode these into issues):

- [ ] When responses are viewed by the survey owner, let them download uploaded files. `process_form_data` could be helpful and presign URLs for this. (Perhaps make `process_form_data` accept a `key_fields_only: bool = False` so as to not do extra work for fields that will be stripped later?)
- [ ] If some absolute crazyperson makes a file upload field a key field for the survey list view, it should probably be a hyperlink to the actual file (presigned URL) as well.

Despite this, superb work. Thank you and please continue ❤️

**Comment by japsu** [comment]
I'll go ahead and merge, then implement alternative number 3 along with #388 #389.

**Pull request merged**